eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -107,13 +107,13 @@ jobs:
             -ProvisionerApplicationSecret '$(provisioner-aad-secret)'
             -Force
             -Verbose
-          condition: ne(variables['AZURE_RESOURCEGROUP_NAME'], '')
+          condition: and(always(), ne(variables['AZURE_RESOURCEGROUP_NAME'], ''))
           continueOnError: true
           displayName: Remove Test Resources
 
 
       - task: PublishTestResults@2
-        condition: always()
+        condition: succeededOrFailed()
```

Comment:
Same comment as above.  I would prefer to keep `always()` for `PublishTestResults`.

Reply:
```succeededOrFailed()``` is the same as ```always()``` except that ```always()``` will keep the job running in the event of manual cancellation. When we manually cancel I am assuming we are expressly want the job cancelled now. I don't see the value of partial test results when we explicitly cancel.

Reply:
The value is this scenario.  Someone notices a test is hung, and would like to manually cancel to stop wasting agent time.  However, they would still like to publish the test results they have so far.  I agree this is a relatively rare scenario, however the cost of delaying cancellation a few seconds (21 seconds for a recent run of `java - storage - tests`) is also low.

Reply:
That's fair enough. Given this is only for the ```- tests``` runs then I guess it won't be a big issue.

Reply:
I'll update the PR, unless you see any other issues we should try to get this merged.